tests/test_subroutine.py

```diff
@@ -1239,10 +1239,11 @@ def test_subroutine_interface(here, frontend):
     Test auto-generation of an interface block for a given subroutine.
     """
     fcode = """
-subroutine test_subroutine_interface (in1, in2, out1, out2)
+subroutine test_subroutine_interface (in1, in2, in3, out1, out2)
```

Comment:
Yes, good catch, although (as you rightly reasoned), changing the capitalisation of the dummy has no effect, because the dummy list is derived from the declaration names, and thus always matches. To my mind, I think this is fine and can go in as is (changing the dummy has no effect, but changing the declaration name changes the dummy name in the test comparison string, which makes the test look awkward).

It does, however, beg the question if our `Array` symbols should expose the array base name as a symbol, and not just a string to inherit case-agnostic comparison, but that is a follow-on issue, I think.

Reply:
OK, sounds good and can go in as is. Thanks for confirming.

Changing the `Array` behaviour would have quite wide-reaching implications because `Array` is in fact just a metasymbol for `ArraySubscript(VariableSymbol('name'))`. And the outward facing behaviour is already mimicking what `VariableSymbol` does, i.e. 
* `Array.name` returns the string, 
* `str(Array())` returns the name plus dimensions (which is the expected behaviour for the nesting with a subscript imho), 
* `Array.symbol` returns the nested `VariableSymbol`
But I might have misunderstood and happy to discuss further.